src/Pages/IncomeGraph.cshtml

```diff
@@ -0,0 +1,55 @@
+﻿@page
+@using System.Globalization
+@using ePiggyWeb.Utilities
+@model ePiggyWeb.Pages.IncomeGraphModel
+
+<form method="get" class="mt-5">
+    <input type="hidden" name="handler" value="Filter" />
+    <input for="startDate" asp-for="StartDate" type="date" value="@Model.StartDate.ToShortDateString()" />
+    <input for="endDate" asp-for="EndDate" type="date" value="@Model.EndDate.ToShortDateString()" />
+    <input type="submit" value="Filter" class="btn btn-warning ml-5" />
+</form>
+
+<div class="row mt-3 mb-5 chart-container">
+    <div class="col-sm" id="graphPie"></div>
+    <div class="col-sm" id="graphColumn"></div>
+</div>
+
+
+
+
+<script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
+<script type="text/javascript">
+    // Load google charts
+    google.charts.load('current', { 'packages': ['corechart'] });
+    google.charts.setOnLoadCallback(drawChart);
+
+    // Draw the chart and set the chart values
+    function drawChart() {
+        var data = google.visualization.arrayToDataTable([
+            ['Importance', 'Amount'],
+            ['Necessary', @Model.Income.GetBy(Importance.Necessary).GetSum().ToString(CultureInfo.InvariantCulture)],
+            ['High', @Model.Income.GetBy(Importance.High).GetSum().ToString(CultureInfo.InvariantCulture)],
+            ['Medium', @Model.Income.GetBy(Importance.Medium).GetSum().ToString(CultureInfo.InvariantCulture)],
+            ['Low', @Model.Income.GetBy(Importance.Low).GetSum().ToString(CultureInfo.InvariantCulture)],
+            ['Unnecessary', @Model.Income.GetBy(Importance.Unnecessary).GetSum().ToString(CultureInfo.InvariantCulture)]
+        ]);
+
+        // Optional; add a title and set the width and height of the chart
+        var options = {
+            'title': 'Income by importance',
+            'height': 600,
+            'vAxis': { 'textStyle': { 'color': '#FFF' }, 'gridlines': { 'count': 5 } },
+            'backgroundColor': '#276dc1',
+            'colors': ['#F44336', '#9C27B0', '#3F51B5', '#2196F3', '#4CAF50'],
+            'legendTextStyle': { color: '#FFF' },
+            'titleTextStyle': { color: '#FFF' },
+            'hAxis': { 'textStyle':{'color': '#FFF'} }
+        };
+
+        var chartPie = new google.visualization.PieChart(document.getElementById('graphPie'));
+        var chartColumn = new google.visualization.ColumnChart(document.getElementById('graphColumn'));
+        chartPie.draw(data, options);
+        chartColumn.draw(data, options);
+    }
+</script>
```

Comment:
Very noticable red symbol says "No newline at end of file" made me wonder if we have and should we have any standard for this or anything?

Reply:
Could you specify where is the red symbol, because for me it doesn't show anything.

Reply:
It was just after the text on line 55, though it doesn't show it here.

Reply:
It was just after the text on line 55, though it doesn't show it here.